src/modules/SX127x/SX1278.cpp

```diff
@@ -471,6 +471,20 @@ int16_t SX1278::explicitHeader() {
   return(setHeaderType(RADIOLIB_SX1278_HEADER_EXPL_MODE));
 }
 
+int16_t SX1278::setDIOMapping(RADIOLIB_PIN_TYPE pin, uint8_t value) {
+  if (pin > 5)
+    return RADIOLIB_ERR_INVALID_DIO_PIN;
+
+  if (pin < 4)
+    return(_mod->SPIsetRegValue(RADIOLIB_SX1278_REG_DIO_MAPPING_1, value, 7 - 2 * pin, 6 - 2 * pin));
+  else
+    return(_mod->SPIsetRegValue(RADIOLIB_SX1278_REG_DIO_MAPPING_2, value, 15 - 2 * pin, 14 - 2 * pin));
+}
+
+int16_t SX1278::setDIOPreambleDetect(bool usePreambleDetect) {
```

Comment:
Is it necessary to have this as a separate method? Seems very specific compared to the other generic methods.

Reply:
This is used to set bit 1 of `RegDioMapping2` which in turn controls what signal is effectively present on the DIO pin when using one of the `RSSI_PREAMBLE_DETECT` values. 
For instance, if you use `RADIOLIB_SX1278_DIO1_CONT_RSSI_PREAMBLE_DETECT` then DIO1 will output either "RSSI threshold reached" or  "Preamble detected" depending on bit 1 of `RegDioMapping2`
So yes, it is specific to the `SX1278` but not in the sense that there is a special method for a given DIO signal, rather a specific method to activate a specific function from the chip itself.